### PR TITLE
WIP [1608] Jinja conversion/refactor

### PR DESCRIPTION
**Pull request by davidread**
- refactoring of package edit form. Reduced use of c in the form. Revamped tests.

### CODE REVIEW

File: ckanext/dgu/tests/functional/test_package_form.py [hunk 1]
```diff
@@ -1,1115 +1,472 @@
...
+from routes import url_for
 
-"""
-from functools import partial
-import json
-import re
+import ckan.new_tests.helpers as helpers
+import ckan.new_tests.factories as factories
+import ckan.model as model
+from ckanext.dgu.tests.functional.base import DguFunctionalTestBase
```
Comment (morty) [inline ckanext/dgu/tests/functional/test_package_form.py]: @davidread did you forget to check in ckanext/dgu/tests/functional/base.py ?

**Comment by davidread** [comment]
base.py is now added - thanks @morty

File: ckanext/dgu/theme/templates/package/new_and_edit.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    window.DGU_RESOURCE_FORMATS = {{h.get_resource_formats()|safe}};
+  </script>
+  <script type="text/javascript" src="/scripts/dgu-package-form.min.js"></script>
+  <link rel="stylesheet" type="text/css" href="/css/jquery-ui-1.9.2.custom.datepicker.min.css" />
+  <script type="text/javascript" src="/scripts/vendor/jquery-ui-1.9.2.custom.datepicker.min.js"></script>
+{% endblock %}
+
+{% block primary_content_inner %}
+  {#- NB c is passed in, but deprecated #}
+  {{- h.snippet(form_snippet, c=c, **form_vars) -}}
```
Comment (morty) [inline ckanext/dgu/theme/templates/package/new_and_edit.html]: @davidread where is form_vars coming from?

Comment (davidread) [inline ckanext/dgu/theme/templates/package/new_and_edit.html]: It comes from this new bit in the controller, copied from ckan master: https://github.com/datagovuk/ckan/pull/5/files#diff-d61fa02346248a2c5eed362244515307R557
Sorry, I should have mentioned this PR relies on this other PR: https://github.com/datagovuk/ckan/pull/5

**Pull request merged**